### PR TITLE
Fix font rendering when rendering multiple PDFs in parallel

### PR DESCRIPTION
**Pull request by Rob--W**
This is work-in-progress. I have only added (failing) unit tests, which shows that PDF.js does not support rendering of multiple PDFs in parallel, as reported in #6205.
- [x] Create unit tests
- [ ] Fix the issue

### CODE REVIEW

**Comment by Rob--W** [comment]
It appears that #5984 fixes this issue. I recommend to merge this unit test after #5984 is merged.

**Comment by timvandermeij** [comment]
Note that that PR is currently blocked by another issue. It would be better to implement the solution in that blocking issue instead of the existing PR as Yury mentioned that the PR is not the best solution.

File: test/unit/api_spec.js [hunk 1]
```diff
@@ -261,4 +261,68 @@ describe('api', function() {
...
+    // Fulfills the promise with the base64-encoded version of the PDF.
+    function renderPDF(filename) {
+      return PDFJS.getDocument(filename)
+        .then(function(pdf) { return pdf.getPage(1); })
+        .then(function(page) {
+          var c = document.createElement('canvas');
+          var v = page.getViewport(1.2);
+          c.width = v.width;
+          c.height = v.height;
+//          document.body.appendChild(c);
```
Comment (timvandermeij) [inline test/unit/api_spec.js]: Can we remove this commented line?

**Comment by timvandermeij** [comment]
#6571 fixes the original issue properly, but when that patch lands the unit tests in this PR are a very useful addition. When that happens, could you rebase this patch and fix the abovementioned comment?

**Comment by timvandermeij** [comment]
#6571 has been merged. Could you update this PR, Rob?

**Comment by Rob--W** [comment]
@timvandermeij Rebased.

**Comment by timvandermeij** [comment]
@Rob--W Could you fix these test failures? I think this is good to go after that.

**Comment by Rob--W** [comment]
@timvandermeij The test passed for me (locally), but I did see an error message in the console (about `FontLoader.clear` not being defined). I've fixed this error, can you retry?

File: test/unit/api_spec.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* globals PDFJS, expect, it, describe, Promise, combineUrl, waitsFor,
            InvalidPDFException, MissingPDFException, StreamType, FontType,
            PDFDocumentProxy, PasswordException, PasswordResponses,
-           PDFPageProxy, createPromiseCapability */
+           PDFPageProxy, createPromiseCapability, FontLoader */
```
Comment (Snuffleupagus) [inline test/unit/api_spec.js]: Nit: the `FontLoader` global no appears to be unused.

**Comment by yurydelendik** [comment]
I don't think for `/botio unittest` command loads PDF files (from cache), hence "{ name : 'MissingPDFException', message : 'Missing PDF "http://127.0.0.1:50076/test/unit/../pdfs/issue1936.pdf".' }" error.

**Comment by Rob--W** [comment]
I've removed the unused global and linked to a different PDF.

**Comment by yurydelendik** [comment]
Actually, can you change the way you destroy documents? The proper way is to call destory() method, which returns a promise. So you can chain that in sequence test with document creation, and invoke destruction in-parallel after all documents are ready.

**Comment by Rob--W** [comment]
@yurydelendik Done.

**Comment by yurydelendik** [comment]
Sorry, I did not notice that before, the second test depends on first test to run. That's not good for the test autonomousness. Please combine two.

Free to land after the fix and testing. Thank you for the patch.

**Comment by Rob--W** [comment]
@yurydelendik Done. I also moved the cleanup to `afterEach` to make sure that the test ends in a clean state.

Can I trigger botio?

/botio unittest

**Comment by yurydelendik** [comment]
> Can I trigger botio?

hmmm, Let me configure that.

/botio unittest

**Comment by yurydelendik** [comment]
@Rob--W try

**Comment by Rob--W** [comment]
Thanks @yurydelendik :)

**Comment by timvandermeij** [comment]
Thank you for the patch!

**Pull request merged**